src/test/functional/functional-tests.js

```diff
@@ -872,8 +899,55 @@ describe('functional tests', function() {
         .catch(() => done())
     })
 
-    step('removeObject(bucketName, objectName, done)__', done => {
-      client.removeObject(bucketName, _1byteObjectName, done)
+    step('generatePresignedUrl(listObjectMethod, bucketName, \'\', expires, reqParams, cb)__', done => {
+      client.generatePresignedUrl('GET', bucketName, '', 1000, {'prefix': 'data', 'max-keys': 1000}, (e, presignedUrl) => {
+        if (e) return done(e)
+        var transport = http
+        var options = _.pick(url.parse(presignedUrl), ['hostname', 'port', 'path', 'protocol'])
+        options.method = 'GET'
+        options.headers = {
+        }
+        var str = ''
+        if (options.protocol === 'https:') transport = https
+        callback = function (response) {
+          if (!/20\d/.test(response.statusCode)) return done(new Error(`error on put : ${response.statusCode}`))
+          response.on('error', e => done(e))
+          response.on('end', function () {
+            parser = new DOMParser();
+            doc = parser.parseFromString(str, 'text/xml');
+            listedObj = doc.getElementsByTagName('Key')[0].childNodes[0];
+            if (_1byteObjectName != listedObj) {
+              return done(new Error('Listed object, '+listedObj+' does not match the object in the bucket, '+_1byteObjectName));
+            }
+            done();
+          });
+          response.on('data', function (chunk) {
+            str += chunk;
+          });
+        }
+        var request = transport.request(options, callback);
+        request.end()
+      })
+    })
+
+    step('generatePresignedUrl(deleteMethod, bucketName, objectName, expires, cb)__', done => {
+      client.generatePresignedUrl('DELETE', bucketName, _1byteObjectName, 1000, (e, presignedUrl) => {
+        if (e) return done(e)
+        var transport = http
+        var options = _.pick(url.parse(presignedUrl), ['hostname', 'port', 'path', 'protocol'])
+        options.method = 'DELETE'
+        options.headers = {
+        }
+        if (options.protocol === 'https:') transport = https
+        var request = transport.request(options, (response) => {
+          if (!/20\d/.test(response.statusCode)) return done(new Error(`error on put : ${response.statusCode}`))
```

Comment:
http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectDELETE.html

looks like delete always returns 204, so you need not worry about 200.

Reply:
Done.